crates/viewer/src/camera.rs

```diff
@@ -11,11 +13,26 @@ enum Projection {
 pub struct Camera {
     projection: Projection,
     aspect_ratio: f32,
+    // Zoom factor used for orthographic projection.
+    zoom_factor: f32,
+    // Position of the camera.
+    position: Vec3,
+    // The upward vector of the camera, determining its orientation.
+    upward: Vec3,
+    // The look-at target, in the center of the view.
+    target: Vec3,
 }
```

Comment:
Talking to Brian, it sounds like the common case is a camera that orbits a target point.
So the behavior is likely easier to implement if the data representation is made for that.

Something like this:
```suggestion
pub struct OrbitCamera {
    projection: Projection,
    aspect_ratio: f32,
    // Zoom factor used for orthographic projection.
    zoom_factor: f32,
    // The look-at target, in the center of the view.
    target: Vec3,
    // The radius of the orbit
    radius: f32,
    // The orientation of the camera around the target point
    orientation: Quat,
}
```

Working with this representation:
- `upward` and `backward` vectors can be extracted from `orientation` on demand
- `position` can be computed on demand as `target + radius * backward`

Reply:
Thank you @mkovaxx! This makes a lot of sense. Let me apply the changes :)

Reply:
@mkovaxx How do I extract `upward` and `backward` from `orientation`. I'm still wrapping my heads around quaternion and glam API.